backend/src/tasks/scheduler.ts

```diff
@@ -241,15 +242,24 @@ const shouldRunScan = async ({
       }
     }
   );
+
   if (
     lastFinishedScanTask &&
     lastFinishedScanTask.finishedAt &&
+    scan.lastRun &&
     lastFinishedScanTask.finishedAt.getTime() >=
       new Date().getTime() - 1000 * scan.frequency
   ) {
     return false;
   }
-
+  if (
```

Comment:
can you add a comment here with something like "should not run a scan when scan is a SingleScan and has already run"?